docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+
+## Yves
+`Yves` application layer provides a lightweight Shop Application.
+
+```
+[Organization]
+└── Yves
+    └── [Module]
+        ├── Controller
+        │   ├── IndexController.php
+        │   └── [Name]Controller.php
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php                
+        ├── Dependency
+        │   ├── Client
+        │   │   ├── [Module]To[Module]ClientBridge.php
+        │   │   └── [Module]To[Module]ClientInterface.php
+        │   ├── Service
+        │   │   ├── [Module]To[Module]ServiceBridge.php
+        │   │   └── [Module]To[Module]ServiceInterface.php           
+        │   └── Plugin
+        │       └── [PluginInterfaceName]PluginInterface.php
+        ├── Plugin
+        │   ├── Provider
+        │   │   ├── [Name]ControllerProvider.php
+        │   │   └── [Name]ServiceProvider.php
+        │   └── [ConsumerModule]
+        │       ├── [PluginName]Plugin.php
+        │       └── [RouterName]Plugin.php
+        ├── Theme
+        │   └── ["default"|theme]
+        │       ├── components
+        │       │   │── organisms
+        │       │   │   └── [organism-name]
+        │       │   │       └── [organism-name].twig
+        │       │   │── atoms
+        │       │   │   └── [atom-name]
+        │       │   │       └── [atom-name].twig
+        │       │   └── molecules
+        │       │       └── [molecule-name]
+        │       │           └── [molecule-name].twig
+        │       ├── templates
+        │       │   ├── page-layout-[page-layout-name]
+        │       │   │   └── page-layout-[page-layout-name].twig
+        │       │   └── [template-name]
+        │       │       └── [template-name].twig
+        │       └── views
+        │           └── [name-of-controller]
+        │               └── [name-of-action].twig        
+        ├── Widget
+        │   └── [Name]Widget.php
+        ├── [Module]Config.php
+        ├── [Module]DependencyProvider.php
+        └── [Module]Factory.php
+```
+
+Used components
+- [Bridge](#bridge)
+- [Config](#module-configurations)
+- [Controller](#controller)
+- [Dependency Provider](#dependency-provider)
+- [Factory](#factory)
+- [Layout](#layout) 
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Provider / Router](#provider--router)
+- [Plugin](#plugin), [Plugin Interface](#plugin-interface)
+- [Templates](#theme)
+- [Theme](#theme)
+- [Widget](#widget)
+
+## Glue
+
+The `Glue` application layer serves as a conduit for providing data access points through APIs.
+It acts as an interface for external systems to interact with the application's data.
+
+```
+[Organization]
+└── Glue
+    └── [Module]
+        ├── Controller
+        │   ├── IndexController.php        
+        │   └── [Name]Controller.php                
+        ├── [CustomDirectory]
+        │   ├── [ModelName]Interface.php
+        │   └── [ModelName].php        
+        ├── Dependency
```

Comment:
Facade also can present in Glue module if it's BackendApi module

Reply:
excellent point!